src/main/java/org/broadinstitute/http/nio/HttpPath.java

```diff
@@ -352,80 +351,73 @@ public Path normalize() {
+        }  else if(other.isAbsolute()){
+            //Note: This violates the general contract of the method but shouldn't be important practically.
+                throw new UnsupportedOperationException("Cannot a resolve an absolute path against an http(s)."
+                        + "\nThis path is: " + this
                         + "\nThe problematic path is an instance of " + other.getClass().getName()
                         + "\nOther path: " + other);
-            }
         } else {
-            final URI otherUri;
             try {
-                otherUri = new URI(other.toString());
+                final URI otherUri = new URI(other.toString());
+                return resolve(otherUri);
             } catch (URISyntaxException e) {
                 throw new IllegalArgumentException("Can only resolve http(s) paths against fully encoded paths which are valid URIs.", e);
             }
-            if(otherUri.getScheme() != null){
-                throw new CantDealWithThisException("Attempting to resolve this against a path which is relatve but looks "
-                        + "like it has a scheme."
-                        + "\nThis: " + this
-                        + "\nOther: " + other
-                        + "\nOther interpretted as URI: " + otherUri
-                        + "\nThis is a limitatation of the current implementation of resolve."
-                        + "\nPlease use choose a less horrible file name or get in touch with the developers to complain.");
-
-            }
-            return new HttpPath(fs, otherUri.getRawQuery(),
-                    otherUri.getRawFragment(),
-                    this.isAbsolute(),
-                    concatPaths(this.normalizedPath, getNormalizedPathBytes(otherUri.getRawPath(), false)));
         }
     }
 
-    /**
-     * Thrown when an input could potentially be handled by this method but it's more
-     * complicated to deal with than the developers had time for
-     */
-    public static class CantDealWithThisException extends IllegalArgumentException{
-        /**
-         * @param message explain why the developers can't deal wih this
-         */
-        public CantDealWithThisException(String message){
-            super(message);
-        }
+    private HttpPath resolve(URI other){
+        return new HttpPath(fs, other.getRawQuery(),
+                other.getRawFragment(),
+                this.isAbsolute(),
+                concatPaths(this.normalizedPath, getNormalizedPathBytes(other.getRawPath(), false)));
     }
 
-
     @Override
     public HttpPath resolve(final String other) {
-        return resolve(other == null ? null: Paths.get(other));
+        return resolve(fromRelativeString(other));
     }
 
     @Override
     public Path resolveSibling(final String other){
-        return resolveSibling(other == null ? null : Paths.get(other));
+        return resolveSibling(fromRelativeString(other));
```

Comment:
Is it worth adding a comment mentioning that we can't use `Paths.get()` here and must go through the `fromRelativeString()` codepath?